File: doc/codes/python/examples/api/command/phase-correction-example.py
import asyncio

from autd3 import Client, ClientConfig
from autd3.commands import SetPhaseCorrection
from autd3.geometry import Autd3, Geometry
from autd3.value import Phase
from autd3_link_nop import Nop


async def main() -> None:
    geometry = Geometry([Autd3([0.0, 0.0, 0.0], [1.0, 0.0, 0.0, 0.0])])
    client = await Client.open(geometry, Nop(), ClientConfig())

    phases = [
        [Phase.ZERO] * geometry.device(i).num_transducers()
        for i in range(geometry.num_devices())
    ]

    builder = client.datagram_builder()
    builder.push(SetPhaseCorrection(phases=phases))
    frames = builder.build()
    for frame in frames:
        await client.send_checked(frame)

    await client.close()


asyncio.run(main())

File: doc/codes/csharp/examples/api/command/phase-correction-example.cs
using System.Numerics;
using System.Threading.Tasks;
using AUTD3;
using AUTD3.Link;
using Nop = AUTD3.Link.Nop;

// HIDE
namespace DocSamples.ApiCommandPhaseCorrectionExample;

internal static class Sample
{
    internal static async Task Run()
    {
        // HIDE_END
var geometry = new Geometry(new[] { new Autd3(Vector3.Zero) });
var client = await Client.OpenAsync(geometry, new Nop(), new ClientConfig());

var phases = new Phase[geometry.NumDevices][];
for (var i = 0; i < geometry.NumDevices; i++)
{
    var count = geometry[i].NumTransducers;
    phases[i] = new Phase[count];
    for (var t = 0; t < count; t++)
    {
        phases[i][t] = Phase.Zero;
    }
}

var builder = client.DatagramBuilder();
builder.Push(new SetPhaseCorrection(phases));
var frames = builder.Build();
foreach (var frame in frames)
{
    await client.SendCheckedAsync(frame);
}

await client.CloseAsync();
        // HIDE
    }
}
// HIDE_END
